### ISSUE TITLE
Proposal for command module error if special characters passed

### ISSUE THREAD

**Issue by joshmullis** [ansible/ansible#22026]
_From @joshmullis on August 26, 2016 1:10_

##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

command module
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### CONFIGURATION

This situation deals with the act of changing the "module_name" config setting.

Day1:   "module_name = shell"
Day2:   "module_name = command"
##### OS / ENVIRONMENT

RHEL 7.2
##### SUMMARY

Proposing the command module errors out when special/invalid characters are passed in the command string as there is a situation where undesirable changes can be made to systems.
##### STEPS TO REPRODUCE

1. Change the default module to "shell" and run the following command.
2. Change the default module back to the default of "command" and re-run the same command.
   Bye-bye /bin/ls binary.  :)

```
ansible all -i inventory -u ansible -s -a "/bin/rm -f /somefile ; /bin/ls /somefile"
```

##### EXPECTED RESULTS

To help avoid a possibly bad situation, I'm proposing the module return an error when invalid characters are passed in...

```
ansible all -i inventory -u ansible -s -a "/bin/rm -f /somefile ; /bin/ls /somefile"
testserver01 | FAILED | rc=60 >>
[Errno 60] Invalid characters passed in command string
```
##### ACTUAL RESULTS

```
ansible all -i inventory -u ansible -s -a "/bin/rm -f/somefile ; /bin/ls /somefile"
testserver01 | SUCCESS | rc=0 >>
```

Since the command module passes special characters as literal, the resulting command would be equivalent to "/bin/rm -f /somefile \; /bin/ls /somefile", which will remove the /bin/ls binary.

While I COMPLETELY understand this is how the command module works (and the shell module should instead be used), there is still an opportunity for accidents.

Thanks!

_Copied from original issue: ansible/ansible-modules-core#4548_

**Comment by joshmullis** [comment]
_From @ansibot on August 26, 2016 1:14_



@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by joshmullis** [comment]
_From @ansibot on September 12, 2016 16:57_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by joshmullis** [comment]
_From @ansibot on October 5, 2016 15:10_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by joshmullis** [comment]
_From @ansibot on December 7, 2016 18:8_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by abadger** [comment]
I don't believe this is doable.  The problem is that the characters aren't really illegal for command.

For instance, say I created a file with a semicolon in it by accident:

```
ansible myhosts -m file -a 'state=touch dest=/tmp/bugs-on-;eopard'
```

I realize I've made the mistake and use command to remove the file:

```
ansible myhosts -m command -a 'rm -f /tmp/bugs-on;eopard'
```

The semicolon is valid there.

Note, in terms of removing files, it's probably better to use file with state=absent (although I know that doesn't address the more generic issue).
